### DISCUSSION TITLE
Error when adding AWS environment variable

### DISCUSSION THREAD

**Discussion by fsa317** [vercel/community#1789] [Help]
### Summary

As soon as I add an AWS environment variable (e.g. AWS_SECRET_ACCESS_KEY) and run vercel dev locally, I get the following error:

```
TypeError: Object prototype may only be an Object or null: undefined
    at Function.setPrototypeOf (<anonymous>)
    at new ValidationError (/usr/local/lib/node_modules/vercel/dist/index.js:8720:16)
    at /usr/local/lib/node_modules/vercel/dist/index.js:8740:25
    at Generator.next (<anonymous>)
    at fulfilled (/usr/local/lib/node_modules/vercel/dist/index.js:8671:58)
```

The stack trace shows logic trying to prevent reserved env variables.  What am I doing wrong?

**Comment by amyegan** [comment]
Hi @fsa317! The error message seems to indicate that the value is undefined. The first thing to check when [using AWS SDK Environment Variables](https://vercel.com/guides/how-can-i-use-aws-sdk-environment-variables-on-vercel) (or any other environment variables) is that you have set the appropriate values for all environments. If it's missing for your development environment, the you could get an undefined value like this. 

After updating environment variables in project settings, you'll also need to run [`vercel pull`](https://vercel.com/docs/cli/pull) to to update your local copy of the environment variables and project settings

**Reply by fsa317** [reply]
After a bunch of debugging I found that the issue is that when my python serverless function is being called from the /api directory this error occurs if I have the AWS env variables set (as they are reserved - this part I dont understand).  If I rename them to something else and use them directly I can get things to work.  I can also get it to work by removing the python functins in /api.  It appears the combination of these reserved env variables and the python serverless function are not playing nice.

For reference my setup is having some other issues - https://github.com/vercel/community/discussions/1698

**Comment by cohlar** [comment]
I confirm the bug. I've been using env variables `AWS_ACCESS_KEY_ID` and `AWS_SECRET_ACCESS_KEY` until now on my web app and nodejs serverless functions without any issue.
Today I added a python serverless function and the below error started to appear. I unchecked "Development" below to make it work locally - this is in contradiction with [the documentation](https://vercel.com/docs/projects/environment-variables/reserved-environment-variables), I would suggest to either create an issue or update the documentation.

```
ValidationError: The following environment variables can not be configured: AWS_ACCESS_KEY_ID, AWS_SECRET_ACCESS_KEY
    at C:\Users\cohla\dev\brandfolio\node_modules\@vercel\fun\dist\src\index.js:93:25
    at Generator.next (<anonymous>)
    at fulfilled (C:\Users\cohla\dev\brandfolio\node_modules\@vercel\fun\dist\src\index.js:24:58) {
  reserved: [ 'AWS_ACCESS_KEY_ID', 'AWS_SECRET_ACCESS_KEY' ]
```

![image](https://github.com/vercel/community/assets/57757008/871c3430-e060-4216-81c3-546f383298c8)
